modules/programs/msmtp.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... } @ top:
+
+with lib;
+with builtins;
+
+let
+  cfg = config.programs.msmtp;
+
+  sendMsmtpCommand = account:
+    if cfg.offlineSendMethod == "native" then
+      "${pkgs.msmtp}/bin/msmtpq -C $XDG_CONFIG_HOME/msmtp/config --account=${account.name} -t"
+    else
+      "${pkgs.msmtp}/bin/msmtp $XDG_CONFIG_HOME/msmtp/config --account=${account.name} -t";
+
+  # msmtp requires the password to finish with a newline
+  passwordCommandStr = account:
+    if account.passwordCommand != null then
+      ''passwordeval bash -c "echo $(${toString account.passwordCommand})"''
+    else "";
+
+  accountStr = {userName, address, realName, ...} @ account: ''
+defaults
```

Comment:
Hmm, won't this produce a defaults section for each account? I would assume it's best to only have one defaults section in the configuration.